src/cascadia/TerminalSettingsEditor/ProfileViewModel.cpp

```diff
@@ -374,6 +391,147 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         BellStyle(currentStyle);
     }
 
+    // Method Description:
+    // - Construct _CurrentBellSounds by importing the _inherited_ value from the model
+    // - Adds a PropertyChanged handler to each BellSoundViewModel to propagate changes to the model
+    void ProfileViewModel::_InitializeCurrentBellSounds()
+    {
+        _CurrentBellSounds = winrt::single_threaded_observable_vector<Editor::BellSoundViewModel>();
+        if (const auto soundList = _profile.BellSound())
+        {
+            for (const auto&& bellSound : soundList)
+            {
+                auto vm = winrt::make<BellSoundViewModel>(bellSound);
+                vm.PropertyChanged({ this, &ProfileViewModel::_BellSoundVMPropertyChanged });
+                _CurrentBellSounds.Append(vm);
+            }
+        }
+        _CurrentBellSounds.VectorChanged([this](auto&&, const IVectorChangedEventArgs& args) {

```

Comment:
Should we really subscribe to our vector if there are only two places that append/remove items respectively? We could just move the corresponding ItemInserted/ItemRemoved code there, no?